src/System.IO.Abstractions.TestingHelpers/FileSystemExtensibility.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+
+namespace System.IO.Abstractions.TestingHelpers
+{
+    internal class FileSystemExtensibility
+    {
+        private readonly Dictionary<string, object> _metadata = new();
+        
+        public virtual void StoreMetadata(string key, object value)
+        {
+            _metadata[key] = value;
+        }
+        
+        public virtual object RetrieveMetadata(string key)
+        {
+            if (_metadata.TryGetValue(key, out object value))
+            {
+                return value;
+            }
+
+            return default;
+        }
+
+        internal void CloneFrom(FileSystemExtensibility template)
```

Comment:
Clone or CloneFrom suggests to me that a new identical object is created.
But this method only copies  the values from the `template`. Any existing metadata are kept.

Perhaps change to `CopyFrom` ?

Reply:
I will change it to "CopyMetadataFrom".